resources/run_sampling.rb

```diff
@@ -268,14 +281,33 @@ def _sample_probability_distribution(prob_dist, num_samples)
       end
     end
 
+    if not (return_samples == return_samples_v2)
```

Comment:
For most cases, the new version is equal to old version; When they are not, making sure that the difference can be justified.